<a href="https://colab.research.google.com/github/cho52o/library_project2019/blob/main/By_University.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np
import pandas as pd

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&scope=email%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdocs.test%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fdrive.photos.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fpeopleapi.readonly&response_type=code

Enter your authorization code:
··········
Mounted at /content/drive


In [ ]:
# 일단 import 
%matplotlib inline  
import matplotlib
import matplotlib as mpl  # 기본 설정 만지는 용도
import matplotlib.pyplot as plt  # 그래프 그리는 용도
import matplotlib.font_manager as fm  # 폰트 관련 용도
import numpy as np, pandas as pd; np.random.seed(0)
import seaborn as sns; sns.set(style="white", color_codes=True)

In [ ]:
print("Matplotlib version", matplotlib.__version__)
%matplotlib inline
%config InlineBackend.figure_format = 'retina'

Matplotlib version 3.1.1


In [ ]:
!apt-get update -qq
!apt-get install fonts-nanum* -qq

Selecting previously unselected package fonts-nanum.
(Reading database ... 132681 files and directories currently installed.)
Preparing to unpack .../fonts-nanum_20170925-1_all.deb ...
Unpacking fonts-nanum (20170925-1) ...
Selecting previously unselected package fonts-nanum-eco.
Preparing to unpack .../fonts-nanum-eco_1.000-6_all.deb ...
Unpacking fonts-nanum-eco (1.000-6) ...
Selecting previously unselected package fonts-nanum-extra.
Preparing to unpack .../fonts-nanum-extra_20170925-1_all.deb ...
Unpacking fonts-nanum-extra (20170925-1) ...
Selecting previously unselected package fonts-nanum-coding.
Preparing to unpack .../fonts-nanum-coding_2.5-1_all.deb ...
Unpacking fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-extra (20170925-1) ...
Setting up fonts-nanum (20170925-1) ...
Setting up fonts-nanum-coding (2.5-1) ...
Setting up fonts-nanum-eco (1.000-6) ...
Processing triggers for fontconfig (2.12.6-0ubuntu2) ...


In [ ]:
path = '/usr/share/fonts/truetype/nanum/NanumGothicEco.ttf'  # 설치된 나눔글꼴중 원하는 녀석의 전체 경로를 가져오자
font_name = fm.FontProperties(fname=path, size=10).get_name()
print(font_name)
plt.rc('font', family=font_name)
fm._rebuild()
mpl.rcParams['axes.unicode_minus'] = False

NanumGothic Eco


In [ ]:
so = pd.read_csv('/content/drive/My Drive/소통형도서관_프로젝트/자료/대출목록.csv')
best = pd.read_csv('/content/drive/My Drive/소통형도서관_프로젝트/자료/베스트_이용도서.csv')

#지난 2018-06-01 ~ 2019-09-30 까지의 대출데이터

In [ ]:
so.head(4)

학과    등록번호  ...        저자       대출일
0  간호학과  569943  ...  하이브로 무사시  20180710
1  간호학과  594738  ...       한비야  20180801
2  간호학과  587517  ...       신동준  20180601
3  간호학과  604495  ...       신재일  20180604

[4 rows x 6 columns]

In [ ]:
so_1 = so.iloc[:, :]
so_2 = so.loc[:,['학과','청구기호','서명','대출일']]

##공과대학 대여권수

In [ ]:
so_eg = so_2[so_2['학과'].isin(['산업경영공학과','컴퓨터공학과','정보통신공학과','멀티미디어공학전공','멀티미디어학부','멀티미디어영상전공',
                                   '토목·건축공학부','미디어영상전공','전기전자공학과' ,'건축공학전공',
                                   '토목환경공학전공','기계공학과','화학공학과','컴퓨터무인통신기술학과'])]

#신규인'신소재공학과','스마트융합공학부'제외 , 재학생이 없을것으로 판단 '공학교육 혁신센터 제외'
so_eg.shape

(1344, 4)

##사범대학 대여권수

In [ ]:
so_edu = so_2[so_2['학과'].isin(['국어교육과','영어교육과','교육학과','역사교육과','미술교육과','수학교육과'])]
#교육연수원, 교직부 제외
so_edu.shape

(2976, 4)

##문과대학 대여권수

In [ ]:
so_la =  so_2[so_2['학과'].isin(['국어국문·창작학과','영어영문학과','외국어문학부','일어일문학전공','프랑스어문학전공',
                                    '문헌정보학과','사학과','기독교학과','인문학부 철학상담학전공','일본·프랑스어문학과'])]
so_la.shape

(4758, 4)

##경상대학 대여권수

In [ ]:
so_ec =  so_2[so_2['학과'].isin(['경영학과','회계학과','무역학과','경제학부','경제학전공',
                                    '중국경제통상학전공','컨벤션호텔경영학과','비즈니스통계학과'])]
#방촌국제화 훈련단 제외
so_ec.shape

(2395, 4)

##법정대학 대여권수

In [ ]:
so_ss =  so_2[so_2['학과'].isin(['법학부','행정·경찰학부' '행정학전공','경찰학전공 모집전공','정치·언론학과' ,'사회복지학과','아동복지학과','도시부동산학과'])]
so_ss.shape

(1778, 4)

##생명나노과학대학 대여권수

In [ ]:
so_ls =  so_2[so_2['학과'].isin(['생명시스템과학과' ,'식품영양학과','화학과' ,'수학과','간호학과','스포츠과학과','화공신소재공학과'])]
so_ls.shape

(2525, 4)

##사회적경제융합대학 대여권수

In [ ]:
so_se =  so_2[so_2['학과'].isin(['사회복지학과','아동복지학과'])]
                               
#상담심리학과 , 사회적경제기업학과 , 빅데이터응용학과 신규 생성학과 제외
so_se.shape

(820, 4)

##린튼글로벌비지니스 대여권수

In [ ]:
so_gb =  so_2[so_2['학과'].isin(['글로벌비즈니스전공','글로벌IT경영전공'])]
so_gb.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 0 entries
Data columns (total 4 columns):
학과      0 non-null object
청구기호    0 non-null object
서명      0 non-null object
대출일     0 non-null object
dtypes: object(4)
memory usage: 0.0+ bytes


##조형예술학부 대여권수

In [ ]:
so_art =  so_2[so_2['학과'].isin(['조형예술학부','융합디자인전공','회화전공','의류학전공'])]
so_art.shape

(860, 4)

##자유전공학부 대여권수

In [ ]:
so_free =  so_2[so_2['학과'].isin(['자유전공학부'])]
so_free.shape

(60, 4)

## 대학별 정리

1)so_eg #공과대

2)so_edu #사범대

3)so_la #문과대

4)so_ec #경상대학

5)so_ss #법정대학

6)so_ls #생명나노대학

7)so_se #사회적경제융합대학

8)so_gb #린튼글로벌비지니스 - 값이 없음 

9)so_art #조형예술학부

10)so_free #자유전공학부


### 데이터 정리

In [ ]:
so_eg.청구기호 = so_eg.청구기호.str.slice(0, 3)
so_eg.대출일 = so_eg.대출일.str.slice(4, 6)
so_edu.청구기호 = so_edu.청구기호.str.slice(0, 3)
so_edu.대출일 = so_edu.대출일.str.slice(4, 6)
so_la.청구기호 = so_la.청구기호.str.slice(0, 3)
so_la.대출일 = so_la.대출일.str.slice(4, 6)
so_ec.청구기호 = so_ec.청구기호.str.slice(0, 3)
so_ec.대출일 = so_ec.대출일.str.slice(4, 6)
so_ss.청구기호 = so_ss.청구기호.str.slice(0, 3)
so_ss.대출일 = so_ss.대출일.str.slice(4, 6)
so_ls.청구기호 = so_ls.청구기호.str.slice(0, 3)
so_se.청구기호 = so_se.청구기호.str.slice(0, 3)
so_gb.청구기호 = so_gb.청구기호.str.slice(0, 3)
so_art.청구기호 = so_art.청구기호.str.slice(0, 3)
so_free.청구기호 = so_free.청구기호.str.slice(0, 3)

/usr/local/lib/python3.6/dist-packages/pandas/core/generic.py:5208: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value


In [ ]:
dso_eg = so_eg.drop(['학과','서명'], axis=1)
dso_edu = so_edu.drop(['학과','서명'], axis=1)
dso_la = so_la.drop(['학과','서명'], axis=1)
dso_ec = so_ec.drop(['학과','서명'], axis=1)
dso_ss = so_ss.drop(['학과','서명'], axis=1)

In [ ]:
dso_eg['청구기호']= dso_eg['청구기호'].astype('int')
dso_eg['대출일'] = dso_eg['대출일'].astype('int')
dso_edu['청구기호']= dso_edu['청구기호'].astype('int')
dso_edu['대출일'] = dso_edu['대출일'].astype('int')
dso_la['청구기호']= dso_la['청구기호'].astype('int')
dso_la['대출일'] = dso_la['대출일'].astype('int')
dso_ec['청구기호']= dso_ec['청구기호'].astype('int')
dso_ec['대출일'] = dso_ec['대출일'].astype('int')
dso_ss['청구기호']= dso_ss['청구기호'].astype('int')
dso_ss['대출일'] = dso_ss['대출일'].astype('int')

### 대학별 도서 선호 장르

1)공과대
  
    1위 813 : 미국문학
    2위 823 : 영국문학
    3위 621 : 공학
    4위 620 : 공학
    5위 006 : 총류


2)사범대
  
    1위 951 : 아시아사
    2위 813 : 미국문학
    3위 370 : 교육학
    4위 810 : 미국문학
    5위 823 : 영국문학


3)문과대
  
    1위 813 : 미국문학
    2위 823 : 영국문학
    3위 895 : 기타 문학
    4위 811 : 미국문학
    5위 843 : 프랑스문학


4)경상대학
  
    1위 658 : 경영 및 보조서비스
    2위 813 : 미국문학
    3위 823 : 영국문학
    4위 895 : 기타 문학
    5위 330 : 경제학


5)법정대학
  
    1위 813 : 미국문학
    2위 895 : 기타 문학
    3위 345 : 법률학
    4위 951 : 아시아사
    5위 361 : 사회문제 및 복지


6)생명나노대학
  
    1위 895 : 기타 문학
    2위 813 : 미국문학
    3위 574 : 생명과학, 생물학
    4위 610 : 의학 및 보건
    5위 820 : 영국문학


7)사회적경제융합대학
  
    1위 813 : 미국문학
    2위 361 : 사회문제 및 복지
    3위 951 : 아시아사
    4위 158 : 심리학
    5위 362 : 사회문제 및 복지


8)린튼글로벌비지니스 - 값이 없음 


9)조형예술학부
  
    1위 745 : 그래픽아트, 장식예술
    2위 741 : 그래픽아트, 장식예술
    3위 813 : 미국문학
    4위 709 : 예술 및 장식 미술
    5위 759 : 회화


10)자유전공학부
  
    1위 813 : 미국문학
    2위 843 : 프랑스문학
    3위 818 : 미국문학
    4위 823 : 영국문학
    5위 895 : 기타 문학


In [ ]:
cso_eg = so_eg.청구기호.value_counts().head(15)
cso_edu = so_edu.청구기호.value_counts().head(15)
cso_la = so_la.청구기호.value_counts().head(15)
cso_ec = so_ec.청구기호.value_counts().head(15)
cso_ss = so_ss.청구기호.value_counts().head(15)
cso_ls = so_ls.청구기호.value_counts().head(15)
cso_se = so_se.청구기호.value_counts().head(15)
cso_gb = so_gb.청구기호.value_counts().head(15)
cso_art = so_art.청구기호.value_counts().head(15)
cso_free = so_free.청구기호.value_counts().head(15)
cso_eg = pd.DataFrame({'청구기호':cso_eg.index, 'count':cso_eg.values})
cso_edu = pd.DataFrame({'청구기호':cso_edu.index, 'count':cso_edu.values})
cso_la = pd.DataFrame({'청구기호':cso_la.index, 'count':cso_la.values})
cso_ec = pd.DataFrame({'청구기호':cso_ec.index, 'count':cso_ec.values})
cso_ss = pd.DataFrame({'청구기호':cso_ss.index, 'count':cso_ss.values})
cso_ls = pd.DataFrame({'청구기호':cso_ls.index, 'count':cso_ls.values})
cso_se = pd.DataFrame({'청구기호':cso_se.index, 'count':cso_se.values})
cso_gb = pd.DataFrame({'청구기호':cso_gb.index, 'count':cso_gb.values})
cso_art = pd.DataFrame({'청구기호':cso_art.index, 'count':cso_art.values})
cso_free = pd.DataFrame({'청구기호':cso_free.index, 'count':cso_free.values})

In [ ]:
cso_eg

In [ ]:
cso_edu

In [ ]:
cso_la

In [ ]:
cso_ec

In [ ]:
cso_ss

In [ ]:
cso_ls

In [ ]:
cso_se

In [ ]:
cso_gb

In [ ]:
cso_art

In [ ]:
cso_free

### 대학별 청구기호와 대출월 관련 그래프

In [ ]:
sns.set_style("whitegrid")
sns.barplot(x="청구기호", y="대출일", data=dso_eg)
plt.suptitle("Engineering", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.barplot(x="청구기호", y="대출일", data=dso_edu)
plt.suptitle("Education", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.barplot(x="청구기호", y="대출일", data=dso_la)
plt.suptitle("Liberal Arts", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.barplot(x="청구기호", y="대출일", data=dso_ec)
plt.suptitle("Business", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.barplot(x="청구기호", y="대출일", data=dso_ss)
plt.suptitle("Law", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=dso_eg)
plt.suptitle("Engineering", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=dso_edu)
plt.suptitle("Education", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=dso_la)
plt.suptitle("Liberal Arts", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=dso_ec)
plt.suptitle("Business", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=dso_ss)
plt.suptitle("Law", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=so_eg, kind="kde")
plt.suptitle("Engineering", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=so_edu, kind="kde")
plt.suptitle("Education", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=so_la, kind="kde")
plt.suptitle("Liberal Arts", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=so_ec, kind="kde")
plt.suptitle("Business", y=1.02)
plt.show()

In [ ]:
sns.set_style("whitegrid")
sns.jointplot(x="청구기호", y="대출일", data=so_ss, kind="kde")
plt.suptitle("Law", y=1.02)
plt.show()

In [ ]:
eg = cso_eg.plot(kind='bar', x='청구기호', y='count', title='Engineering', color=['cornflowerblue', 'lightskyblue','paleturquoise', 'springgreen',  'lightgreen', 'greenyellow','yellow','gold', 'orange', 'coral'], figsize=(12, 4), fontsize=12)
#eg.set_xlabel('서명', fontsize=12)          # x축 정보 표시
#eg.set_ylabel('LB_sum', fontsize=12)     # y축 정보 표시
#eg.legend(['LB_sum'], fontsize=12)    # 범례 지정

In [ ]:
edu = cso_edu.plot(kind='bar', x='청구기호', y='count', title='Education', color=['cornflowerblue', 'lightskyblue','paleturquoise', 'springgreen',  'lightgreen', 'greenyellow','yellow','gold', 'orange', 'coral'], figsize=(12, 4), fontsize=12)

In [ ]:
la = cso_la.plot(kind='bar', x='청구기호', y='count', title='Liberal Arts', color=['cornflowerblue', 'lightskyblue','paleturquoise', 'springgreen',  'lightgreen', 'greenyellow','yellow','gold', 'orange', 'coral'], figsize=(12, 4), fontsize=12)

In [ ]:
ec = cso_ec.plot(kind='bar', x='청구기호', y='count', title='Business', color=['cornflowerblue', 'lightskyblue','paleturquoise', 'springgreen',  'lightgreen', 'greenyellow','yellow','gold', 'orange', 'coral'], figsize=(12, 4), fontsize=12)

In [ ]:
ss = cso_ss.plot(kind='bar', x='청구기호', y='count', title='Law', color=['cornflowerblue', 'lightskyblue','paleturquoise', 'springgreen',  'lightgreen', 'greenyellow','yellow','gold', 'orange', 'coral'], figsize=(12, 4), fontsize=12)